helm/values.yaml

```diff
@@ -0,0 +1,256 @@
+# Default values for authelia.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: clems4ever/authelia
+  tag: v3.16.2
+  pullPolicy: IfNotPresent
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+authelia:
+  logLevel: "debug"
+  redirectionURL: "https://home.example.com:8080/"
+  totpIssuer: "authelia.com"
+
+  duo:
+    enabled: false
+    config:
+      hostname: api-123456789.example.com
+      integration_key: ABCDEF
+      secret_key: 1234567890abcdefghifjkl
+
+  ldap:
+    enabled: false
+    config:
+      url: ldap://127.0.0.1
+      # The base dn for every entries
+      base_dn: dc=example,dc=com
+      # An additional dn to define the scope to all users
+      additional_users_dn: ou=users
+      # The users filter used to find the user DN
+      # {0} is a matcher replaced by username.
+      # 'cn={0}' by default.
+      users_filter: cn={0}
+      # An additional dn to define the scope of groups
+      additional_groups_dn: ou=groups
+      # The groups filter used for retrieving groups of a given user.
+      # {0} is a matcher replaced by username.
+      # {dn} is a matcher replaced by user DN.
+      # {uid} is a matcher replaced by user uid.
+      # 'member={dn}' by default.
+      groups_filter: (&(member={dn})(objectclass=groupOfNames))
+      # The attribute holding the name of the group
+      group_name_attribute: cn
+      # The attribute holding the mail address of the user
+      mail_attribute: mail
+      # The username and password of the admin user.
+      user: cn=admin,dc=example,dc=com
+      password: password
+
+    # If LDAP is disabled, you can specify a users database file, however
+    # it will be read-only!
+  fileAuth:
+    users: {}
+      # john:
+      #   password: "{CRYPT}$6$rounds=500000$jgiCMRyGXzoqpxS3$w2pJeZnnH8bwW3zzvoMWtTRfQYsHbWbD/hquuQ5vUeIyl9gdwBIt6RWk2S6afBA0DPakbeWgD/4SZPiS0hYtU/"
+      #   email: john.doe@authelia.com
+      #   groups:
+      #     - admins
+      #     - dev
+
+  # Default policy can either be `bypass`, `one_factor`, `two_factor` or `deny`.
+  # It is the policy applied to any resource if there is no policy to be applied
+  # to the user.
+  accessControl:
+    defaultPolicy: deny
+    rules:
+      # Rules applied to everyone
+      - domain: public.example.com
```

Comment:
As I said earlier, I think this part could probably be integrated in the ingress configuration to improve integration with Kubernetes. That would make a lot of sense to define everything related to the endpoint in a single place.

Reply:
I mean it's a neat idea, but rules can be quite complicated, and annotations are supposed to be one line each. A CRD would be a better (but more complicated!) approach

Reply:
Yep, CRD is definitely something we can consider in the future.